server-spi/src/main/java/org/keycloak/provider/ProviderFactory.java

```diff
@@ -67,4 +69,13 @@ default int order() {
     default List<ProviderConfigProperty> getConfigMetadata() {
         return Collections.emptyList();
     }
+
+    /**
+     * Optional method used to declare that a ProviderFactory has a dependency on one or more Providers. If a Provider
+     * is declared here, it is guaranteed that the dependencies {@link #postInit} method will be executed
+     * before this ProviderFactory's {@link #postInit}.
+     */
+    default Set<Class<? extends Provider>> dependsOn() {
+        return Set.of();
```

Comment:
Should this not use `Collections.emptySet()`?

Reply:
I don't think we want the returned set to be mutable. `Set.of()`  just calls `ImmutableCollections.EMPTY_SET`.

Reply:
emptySet is immutable as well, but yeah if Set.of() returns the same then it doesn't really matter, an extra indirection though compared to just explicitly saying empty set (which is also a little bit more readable).

Nitpick though ;)